### PR TITLE
2681/listing application back button

### PR DESCRIPTION
**Pull request by adriencyberspace**
# Pull Request Template

## Issue Overview

This PR addresses #2681

- [X] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

This fixes the back button on the /contact/address page of the application listing. Because both the address form and the verify address form share a url, the back button for both views takes you back to the previous url: /contact/name. This fixes the button so that if you are on the verify form page, the back button takes you back to the address form page, but if you are on the address form page, the back button takes you to the previous url. 

## How Can This Be Tested/Reviewed?

1. Click any listing in public and apply online
2. Once you reach the /applications/contact/address page, fill out the form and click next
3. Once on the verify address page, click back
4. You will be taken back to the correct view - the one you just came from. All of your info will still be populated
5. Click back again and you will be on the /applications/contact/name page 

Describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

Entry Point: http://localhost:3000/applications/contact/address
- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Comment by emilyjablonski** [comment]
I'd check out the failing Cypress test, it seems like it may be related

**Comment by adriencyberspace** [comment]
Don't merge! Debugging.

**Comment by adriencyberspace** [comment]
@emilyjablonski Should be fixed now. The problem was in using conductor.currentStep.url instead of window.location.pathname or hardcoding the url. Any idea why that is?

File: sites/public/pages/applications/contact/address.tsx [hunk 1]
```diff
@@ -142,10 +142,11 @@ const ApplicationAddress = () => {
       </FormCard>
       <FormCard>
         <FormBackLink
-          url={conductor.determinePreviousUrl()}
-          onClick={() => conductor.setNavigatedBack(true)}
+          url={verifyAddress ? window.location.pathname : conductor.determinePreviousUrl()}
```
Comment (YazeedLoonat) [inline sites/public/pages/applications/contact/address.tsx:145]: can this use the useMemo hook instead

File: sites/public/pages/applications/contact/address.tsx [hunk 2]
```diff
@@ -142,10 +142,11 @@ const ApplicationAddress = () => {
...
+          onClick={
+            verifyAddress ? () => setVerifyAddress(false) : () => conductor.setNavigatedBack(true)
```
Comment (YazeedLoonat) [inline sites/public/pages/applications/contact/address.tsx:147]: can this use the callback hook instead

**Review by YazeedLoonat — CHANGES_REQUESTED** [review]
Looks good, I think it would be better to switch to using the react hooks for these instead

**Review by YazeedLoonat — APPROVED** [review]
LGTM

**Pull request merged**